integration-tests/src/tests/client/resharding.rs

```diff
@@ -1373,6 +1383,38 @@ fn test_shard_layout_upgrade_missing_chunks(
     test_env.check_split_states_artifacts();
 }
 
+fn test_shard_layout_upgrade_missing_chunks(
+    resharding_type: ReshardingType,
+    p_missing: f64,
+    rng_seed: u64,
+) {
+    init_test_logger();
+
+    let genesis_protocol_version = get_genesis_protocol_version(&resharding_type);
+    let target_protocol_version = get_target_protocol_version(&resharding_type);
+    let epoch_length = 10;
+
+    let mut test_env = create_test_env_for_cross_contract_test(
+        genesis_protocol_version,
+        epoch_length,
+        rng_seed,
+        Some(resharding_type),
+    );
+    test_missing_chunks(&mut test_env, p_missing, target_protocol_version, epoch_length)
+}
+
+// Use resharding setup to run protocol for couple epoch with given probability
+// of missing chunk. In particular, checks that all txs generated in env have
+// final outcome in the end.
+// TODO: remove logical dependency on resharding.
+fn test_latest_protocol_missing_chunks(p_missing: f64, rng_seed: u64) {
```

Comment:
Confirming what's happening... High level, we are running with the latest protocol version, using the infra of the missing chunks setup, NOT doing any sort of resharding and just checking whether we expect the right set of transactions and receipts in case the chunks are missing?

Reply:
I'm sorry, I didn't quite understand how are these tests to evolve in the future?

Reply:
I think the tests will remain roughly the same but everything else around them will change (pre state root -> new pre state root).

Reply:
> Confirming what's happening... High level, we are running with the latest protocol version, using the infra of the missing chunks setup, NOT doing any sort of resharding and just checking whether we expect the right set of transactions and receipts in case the chunks are missing?

Yeah, exactly.